CMakeLists.txt

```diff
@@ -5,11 +5,58 @@ set(PROJECT_NAME Covid19EERAModel)
 project(${PROJECT_NAME} VERSION 0.0.0 LANGUAGES CXX)
 
 set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake_modules)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set (CMAKE_CXX_FLAGS "-std=gnu++11 -Wall -Wextra -DROOT_DIR=\\\"${CMAKE_SOURCE_DIR}\\\"")
+set(CMAKE_BUILD_TYPE Debug)
+set(COVERAGE_COMPILER_FLAGS "-g -fprofile-arcs -ftest-coverage")
+set (CMAKE_CXX_FLAGS "-std=gnu++11 -Wall -Wextra -O0 ${COVERAGE_COMPILER_FLAGS} -DROOT_DIR=\\\"${CMAKE_SOURCE_DIR}\\\"")
 
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 
 add_subdirectory(src)
 add_subdirectory(test)
+
+find_program( LCOV_PATH  NAMES lcov lcov.bat lcov.exe lcov.perl)

```

Comment:
Is it the case that people who do not have these things installed still be able to run the build on their machines, or will this cause their builds to error out?

Reply:
If it fails to find `lcov` then the variable is not set and then `make coverage` would fail. The main build should not fail.

Reply:
I quite like the workflow yml,  for the CMakeLists.txt, I hope for a little tweak

```
if (NOT DEFINED CMAKE_BUILD_TYPE)
    set(CMAKE_BUILD_TYPE Debug)
endif()
```
In the short term, we default to debug, later to Release. Currently, user can overwrite this by  "cmake .. -DCMAKE_BUILD_TYPE=Release" in benchmarking. 

For coverage, we may make an cmake option, default to OFF.    So the user will not even notice there is coverage stuff.     

While this is only my way of doing things,  for your reference.
https://github.com/qingfengxia/cpp_numeric_cast/blob/master/CMakeLists.txt